pkg/openstack/loadbalancer.go

```diff
@@ -2317,6 +2350,12 @@ func (lbaas *LbaasV2) ensureAndUpdateOctaviaSecurityGroup(clusterName string, ap
 			return fmt.Errorf("failed to create Security Group for loadbalancer service %s/%s: %v", apiService.Namespace, apiService.Name, err)
 		}
 		lbSecGroupID = lbSecGroup.ID
+
+		tags := lbaas.getCustomLoadBalancerTags(apiService, svcConf)
+
+		if _, err := neutrontags.ReplaceAll(lbaas.network, "security-group", lbSecGroupID, neutrontags.ReplaceAllOpts{Tags: tags}).Extract(); err != nil {
```

Comment:
Had you tested this and it works? I've only tried with `security-groups` and can confirm it was working as commented here: https://github.com/kubernetes/cloud-provider-openstack/pull/2439#issuecomment-1804183413